### DISCUSSION TITLE
Problem faced

### DISCUSSION THREAD

**Discussion by sivasubm1** [sroebert/mqtt-nio#18] [General]
Hi,

With Mqtt-nio package i am able to successfully present my POC.

Problem Facing:

We need to add this swift code under "IOS" section of a React-native common library.
That React-native common library will be installed as a npm package in another React-native app.

Since react-native is supporting Podspec files , and also react-native app depends on cocoapods dependency
 Is it possible for you to give support for cocoapod install..which may be big help on completing this project..thanks

**Comment by sroebert** [comment]
I'm afraid you will have to look into support for Swift Package Manager. I cannot imagine React Native does not support this. This library depends on other Apple Swift packages which also do not support Cocoapods by default, so I feel like it will be much easier trying to find a way to use Swift Package Manager. Another option might be to create your own Cocoapods library and make that dependent on this library.

**Comment by sivasubm1** [comment]
Hi..i have one qn..

is there any autoreconnect options available..

**Comment by sroebert** [comment]
Yes there is, you can have a look at `reconnectMode` in the configuration of the client.
